src/ipa/rpi/common/ipa_base.cpp

```diff
@@ -1260,6 +1267,22 @@ void IpaBase::applyControls(const ControlList &controls)
 			statsMetadataOutput_ = ctrl.second.get<bool>();
 			break;
 
+		case controls::rpi::SYNC_MODE: {
+			SyncAlgorithm *sync =  dynamic_cast<SyncAlgorithm *>(controller_.getAlgorithm("sync"));
+
+			if (sync) {
+				int mode = ctrl.second.get<int32_t>();
+				SyncAlgorithm::Mode m = SyncAlgorithm::Mode::Off;
+				if (mode == controls::rpi::SyncModeServer)
+					m = SyncAlgorithm::Mode::Server;
+				else if (mode == controls::rpi::SyncModeClient)
+					m = SyncAlgorithm::Mode::Client;
+				sync->setMode(m);
+				LOG(IPARPI, Info) << "Mode set";
```

Comment:
Could we say what mode we set? I think it would be nicer for the user if it was clear what had happened, e.g. "Sync mode set to server"